docs/concepts/manifest-builds.md

```diff
@@ -106,17 +108,18 @@ the build an artifact that runs the `hello` package.
 By setting `runtime-packages = [ "hello" ]` we exclude `ripgrep` from the
 closure of the `hello-pkg` artifact.
 
-## Where to put artifacts
+### Where to put artifacts
 
 To keep the output of a build separate from the source files,
 every build is supplied with a directory whose path is stored in a variable
 named `out`.
 Only the files stored in this directory are considered part of the output of
-a build.
+a build and it is empty by default.
```

Comment:
suggestion(non-blocking): In the Nix world, the reason it is as such is historical. It's so that autoconf-style packages can do something as follows.

```console
$ ./configure \
  --bindir=$out/bin \
  --libdir=$out/lib \
  --includedir=$out/include \
  --docdir=$out/share/doc \
  ...
  ``` 
  
  I'm not sure if this is useful information, or makes the whole thing confusing, but I think the historical reason is cool :).

Reply:
I did wonder in the past (before we had the warnings on empty `$out`) whether we should pre-create directories but it would/could result in unnecessary empty directories in the resulting package which would be confusing and messy. I don't think it's worth including that level of detail in this doc though.